manifests/init.pp

```diff
@@ -151,7 +151,7 @@
   $settings          = {},
   $startup_options   = {},
   $jvm_options       = [],
-  Array $pipelines   = [],
+  Variant[Boolean,Array[Hash,1]] $pipelines = false,
   Boolean $manage_repo   = true,
```

Comment:
Would it be possible to do this without overloading the `$pipelines` variable? Having a variable that can be either an array or a boolean is a bit odd and it kinda goes against Puppet's continuing march toward type safety.

Also, what would happen if I declared an array of pipelines and also added another one as a defined type elsewhere? Can I do that with this implementation?

I think the ideal API would allow:

- 0 or more pipelines passed as an array to the class
- 0 or more pipelines declared as defined types

Ideally, neither of these should require a flag to "turn them on" and they should be able to co-exist. Do you think it's possible?